### PR TITLE
Added Utility Items 

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Create a new gui for the purchasing of Lootbox instead of an addAction.
Added a two that can be purchased, they can be used for refueling. One of them is a small flexible fuel tank with 300 liters.
The other item is a large fuel tanker that holds 10000 liters and can be used to refuel fuel sources. Example. 
If the player is in a vehicle that returns true for ([vehicle player] call HR_GRG_fnc_isFuelSource) and is looking at the fuel tanker they can refuel the fuel cargo. The issue with added the large fuel tanker is what to charge it. I left the price at 2000, but i full expect us to change that price to something higher. I'd like to keep the tanker a constant price. The tanker can't be garaged therefore the player or commander can't get any more money back and we don't have to worry about if we do garage it, do we add it to the current fuel sources. I'd like to not add that functionality to this. Just have the tanker be out in the open like a bunker.  I added a canRefuel function but I don't use it. I plan to use it like Håkon's loot to crate. But not today. It will probably refracted. Also added a parameter like LTC's but for refueling. 

I choose to add a new case statement in the callback hander for the garage. 

Please ask me any questions about this. I have a lot of free time.
    

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

Needs to tested on a dedicated server. With multiple people interacting with the tanker.
********************************************************
Notes:

### CODE REVIEW

**Comment by Bob-Murphy** [comment]
Might will have to wait after the UI update because of the GUI part

File: A3-Antistasi/Stringtable.xml [hunk 1]
```diff
@@ -629,6 +629,50 @@
...
+      <Key ID="STR_antistasi_dialogs_big_jerrycan_buy_tooltip">
+        <Original>A large fuel source for refueling</Original>
+      </Key>
+      <Key ID="STR_antistasi_dialogs_jerrycan_buy">
+        <Original>Jerry Can</Original>
+      </Key>
+      <Key ID="STR_antistasi_dialogs_jerrycan_buy_tooltip">
+        <Original>A simple Jerry Can for refueling</Original>
+      </Key>
+
```
Comment (Bob-Murphy) [inline A3-Antistasi/Stringtable.xml:672]: whitespaces bad, please remove

Comment (killerswin2) [reply A3-Antistasi/Stringtable.xml]: done

**Comment by HakonRydland** [comment]
As the ui rework will have a way to buy loot crates, the jerry cans can be added there quite simply, the large tanker seems excessive and will likely not be accepted.
in general you need: 
- better ace safeties in each ace dependant function.
- remove unused stuff (commented out code unused functions etc...)
- fix formatting (weird indentations) see A3A dev guidelines in the wiki.
- scope safety with variables( private _var etc...).
- make everything templatable (no static classnames).
   - under this you also need to make sure its making it a fuel source when its not already one, maybe also setting it to a consistent quantity).
- as your thing is ace dependant, dont use the garage isSource, use the ace functions to determine, the garage source detection is not ace exclusive and could give false positives.
- other minor quality improvements
   - bool return functions dont need to be compared against another bool in a condition check, its already a bool.
   - `==` is less safe and efficient than `isEqualTo` for the checks you are doing.
   - probably more, but you have time to review it yourself.
   
as you likely have a bit of time before the UI reworki is merged, i recommend looking into cleaning up, ensuring safety, and improving efficiency and readability of your code outside the UI.
when the UI rework is merged you can come back to implementing it into the new ui

**Comment by killerswin2** [comment]
Gonna close this, i'll reopen once I take care of the issues that Hakon raised, and once the new ui is in effect. Thank y'all for your time. I'll try to be more productive with everyone's time next time.

**Comment by HakonRydland** [comment]
Thank you for your time and effort, look forward to the next PR

**Pull request closed without merging**